.github/workflows/tag.yml

```diff
@@ -1,15 +1,16 @@
 name: Tag
 
 on:
-  push:
-    branches: ["main"]
-  workflow_dispatch:
+  pull_request:
+    branches:
+      - 'main'
+      - 'patch-*'
```

Comment:
This workflow is now split into two sequential jobs:

1. `generate-tag` - Parses the VERSION.yaml into a tag string such as `v1.2.0` and validates that the parsed tag string adheres to expectations, namely:
    - Does not already exist on the repo
    - Starts with `v` and contains three dot-delimited semantic elements
    - If this is a PR to a `patch-` branch, the third semantic element should _not_ be zero
2. `cut-release` - If this PR was merged, actually go ahead and cut the new release

I have changed the trigger to `pull_request` so that we can run step `1` (ensuring validation of tags) before merge.

PRs to branches `main` (for regular develpopment) and those beginning with `patch-` (for patches) will get this workflow to run. PRs to other branches will not trigger this,